File: python/findPrimes.py

import math

def SieveOfEratosthenes(n): 
       
    primes = []
    b = [True for i in range(n + 1)] 
    i = 2
    
    b[0] = False
    b[1] = False

    while (i * i <= n): 
        if (b[i] == True): 
            for j in range(i * 2, n + 1, i): 
                b[j] = False
        i += 1

    for i in range(n + 1): 
        if b[i]: 
            primes.append(i)
    return primes
  
if __name__=='__main__': 
    primes = SieveOfEratosthenes(10000000)
    print(len(primes))

        

File: golang/findPrimes.go
package main

func sieveOfEratosthenes(n int) (primes []int) {
	b := make([]bool, n)

	for i, _ := range b {
		b[i] = true
	}
	for i := 2; i*i < n; i++ {
		if b[i] == true {
			for j := i * i; j < n; j = j + i {
				b[j] = false
			}
		}
	}
	for i := 2; i < len(b); i++ {
		if b[i] == true {
			primes = append(primes, i)
		}
	}
	return primes
}

func main() {
	primes := sieveOfEratosthenes(10000000)
	println(len(primes))
}
